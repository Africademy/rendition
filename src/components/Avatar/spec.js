/* globals expect, describe, it */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { Avatar, Provider } from '../../../dist'

describe('Avatar component', () => {
  it('Avatar renders correctly', () => {
    const component = renderer.create(
      <Provider>
        <Avatar />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('firstName and lastName properties', () => {
    it('should render the Avatar with initials if src is not passed', () => {
      const component = mount(
        <Provider>
          <Avatar firstName='test' lastName='user' />
        </Provider>
      )
      expect(component.text()).toContain('TU')
    })
  })

  describe('null src, valid firstname and valid lastName properties', () => {
    it('should render the Avatar with initials if src is passed but it is null', () => {
      const component = mount(
        <Provider>
          <Avatar src={null} firstName='example' lastName='user' />
        </Provider>
      )
      expect(component.text()).toContain('EU')
    })
  })
})
