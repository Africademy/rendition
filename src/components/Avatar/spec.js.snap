// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component Avatar renders correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 48px;
  width: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 100%;
  overflow: hidden;
}

.c0 {
  font-family: 'Source Sans Pro',Arial,sans-serif;
  font-size: 14px;
  line-height: 1.5;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-family: 'Source Sans Pro',Arial,sans-serif;
  font-size: 14px;
  color: #2A506F;
}

<div
  className="c0 c1"
>
  
  <div
    className="c2 "
  />
</div>
`;
